code/game/objects/items/devices/radio/headset.dm

```diff
@@ -113,6 +114,84 @@ GLOBAL_LIST_INIT(channel_tokens, list(
 	icon_state = "sec_headset"
 	worn_icon_state = "sec_headset"
 	keyslot = new /obj/item/encryptionkey/headset_sec
+	actions_types = list(/datum/action/item_action/dispatch)
+	var/radio_channel = RADIO_CHANNEL_SECURITY
+	COOLDOWN_DECLARE(dispatch_cooldown_timer)
+	var/dispatch_cooldown = 20 SECONDS
+
+/obj/item/radio/headset/headset_sec/ui_action_click(mob/user, action)
+	if(istype(action, /datum/action/item_action/dispatch))
+		dispatch(user)
+	else
+		to_chat("What else is there, moron?")
+
+GLOBAL_LIST_EMPTY(secsets)
+
+// **** Dispatch ****
+
+/datum/action/item_action/dispatch
+	name = "Signal dispatch"
+	desc = "Opens up a quick select wheel for reporting crimes, including your current location, to your fellow security officers."
+	button_icon_state = "dispatch"
+	icon_icon = 'icons/mob/actions/hailer_actions.dmi'
+
+/obj/item/radio/headset/headset_sec/Destroy()
+	GLOB.secsets -= src
+	. = ..()
+
+/obj/item/radio/headset/headset_sec/Initialize(mapload)
+	. = ..()
+	GLOB.secsets += src
+	//recalculateChannels()
+
+/obj/item/radio/headset/headset_sec/proc/dispatch(mob/living/user)
+
+	if(COOLDOWN_TIMELEFT(src, dispatch_cooldown_timer))
+		to_chat(user, "<span class='warning'>Dispatch radio broadcasting systems are recharging.</span>")
+		balloon_alert(user, "still recharging!")
+		return FALSE
+	var/list/options = list()
+	for(var/option in list(
+		"Requesting Backup",
+		"Detaining Perp",
+		"Awaiting Orders",
+		"Falling Back",
+		))
+		//Hardcoded for each icon, not all crimes need emergency callout for more officers
+		options[option] = image(icon = 'icons/effects/aiming.dmi', icon_state = option)
+
+	var/message = show_radial_menu(user, user, options)
+	if(!message)
+		return FALSE
+
+	var/area/current_area = get_area(user)
+	var/sanitized_area_string
+	if(!current_area)
+		sanitized_area_string = "NULL"
+	else if(istype(current_area, /area/crew_quarters))
+		sanitized_area_string = "Crew Quarters"
+	else if(istype(current_area, /area/medical))
+		sanitized_area_string = "Medical"
+	else if(istype(current_area, /area/science))
+		sanitized_area_string = "Science"
+	else if(istype(current_area, /area/engineering))
+		sanitized_area_string = "Engineering"
+	else if(istype(current_area, /area/bridge))
+		sanitized_area_string = "Bridge"
+	else if(istype(current_area, /area/security))
+		sanitized_area_string = "Security"
+	else
+		sanitized_area_string = "UNKNOWN"
```

Comment:
The following areas should probably show up as something:
- Maintenance (All show as one area to make it difficult to find someone reporting for help in maints)
- Cargo bay

Instead of unknown you should try to find something that would make sense if spoken, since unknown will probably appear quite frequently and shouldn't seem like a bug.

Reply:
bc78891